src/tiled/moveselectiontool.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * moveselectiontool.h
+ * Copyright 2017, Ketan Gupta <ketan19972010@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "abstracttiletool.h"
+#include "tilelayer.h"
+
+namespace Tiled {
+namespace Internal {
+
+class MoveSelectionTool : public AbstractTileTool
+{
+    Q_OBJECT
+
+public:
+    MoveSelectionTool(QObject *parent = nullptr);
+    ~MoveSelectionTool();
+
+    void activate(MapScene *scene) override;
+    void deactivate(MapScene *scene) override;
+
```

Comment:
You'll want to override the default implementations of `mouseEntered` and `mouseLeft`, since they're showing/hiding the preview item, which is not what you want for this tool.